service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/ClientInstallationServiceImpl.java

```diff
@@ -83,10 +83,17 @@ public void addInstallation(Variant variant, Installation entity) {
 
     private void mergeCategories(Installation entity) {
         if (entity.getCategories() != null) {
-            List<String> categoryNames = convertToNames(entity.getCategories());
-            final List<Category> categories = categoryDao.findByNames(categoryNames);
-            entity.getCategories().removeAll(categories);
-            entity.getCategories().addAll(categories);
+
+            if(!entity.getCategories().isEmpty()) {
+                final List<String> categoryNames = convertToNames(entity.getCategories());
+                final List<Category> categories = categoryDao.findByNames(categoryNames);
+                entity.getCategories().removeAll(categories);
+                entity.getCategories().addAll(categories);
+            }
+            //Maybe with the update, categories has all been removed
+            else {
+                entity.getCategories().clear();
```

Comment:
Because even if there are no  categories, `getCategories` is not null (at least with postgres/mysql) but empty. The `else` clause is here to handle the case we update the registration with 0 categories.